extensions/binding/org.eclipse.smarthome.binding.tradfri/src/main/java/org/eclipse/smarthome/binding/tradfri/internal/TradfriCoapEndpoint.java

```diff
@@ -0,0 +1,36 @@
+/**
+ * Copyright (c) 2014-2017 by the respective copyright holders.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.binding.tradfri.internal;
+
+import org.eclipse.californium.core.network.CoapEndpoint;
+import org.eclipse.californium.core.network.config.NetworkConfig;
+import org.eclipse.californium.scandium.DTLSConnector;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class wraps {@link CoapEndpoint} from californium for the sole purpose of adding some debug logging to it in
+ * order to figure out when the endpoint is destroyed.
+ * See https://github.com/eclipse/californium/pull/452#issuecomment-341703735
+ *
+ * @author Kai Kreuzer - Initial contribution
+ */
+public class TradfriCoapEndpoint extends CoapEndpoint {
+
+    private final Logger logger = LoggerFactory.getLogger(TradfriCoapEndpoint.class);
+
+    public TradfriCoapEndpoint(DTLSConnector dtlsConnector, NetworkConfig standard) {
+        super(dtlsConnector, standard);
+    }
+
+    @Override
+    public synchronized void destroy() {
+        logger.debug("Destroying CoAP endpoint.", new Exception("Endpoint destroyed"));
```

Comment:
Shouldn't we add an enclosing "if is debug log is enabled"?
The exception instance is created regardless of the log level and so the class instance needs already to catch the whole stack trace etc. Isn't the creation of an exception and its stack trace array building a very expensive operation?
The exception is only for logging so it does not really care which class is used, but I still prefer a runtime one :wink: (if you don't care).

Reply:
Yes, it is potentially expensive, but it is a piece of code that ideally should hardly ever be executed.
But your comment is correct in general, so I have updated the PR as suggested.